_posts/guides/2022-04-29-guide_porting-goldsrc-to-source.md

```diff
@@ -17,36 +17,123 @@ toc: true
+The pros and cons of this decompiler are as follows:  
+- Exclusively Face-To-Brush decompiles 
+- Has not been updated since the early 2000s
+- Command line only, no GUI
+- Does not typically crash when decompiling
+- Prone to generating [spaghetti](https://i.imgur.com/1mIOKCV.png)
+- Limited decompile customization options
+- Can extract embedded textures
 
-### Decompiling with WinBSPC
 
-Open [WinBSPC](https://valvedev.info/tools/winbspc/), in the top left corner select file > convert. Navigate to the BSP file you wish to decompile and select it. Choose Convert to: MAP, and best match texturing. Choose the desired output folder and click OK.
+### Decompiling with WinBSPC
 
-![DECOMPILE PARAMETERS](/assets/images/goldsrc_to_source_guide/WINBSPC_PARAMETERS.png)
+Open [WinBSPC](https://valvedev.info/tools/winbspc/), in the top left corner select file > convert. Navigate to the BSP file you wish to decompile and select it. Select the "MAP" option under "Convert to" and make sure "best match texturing" is not selected under "MAP options". Choose the desired output folder and click OK.
 
 ### Decompiling with BSPTwoMap
 
 [BSPTwoMap](https://valvedev.info/tools/bsptwomap/), is a command-line application, simply drag the bsp onto to .exe to decompile it.
 
 ## Converting the textures
 
-Some maps don't embed textures and simply come with a .WAD however most do not. As mentioned earlier, BSPTwoMap will extract them on decompile, it's recommended to simply decompile the BSP again with BSPTwoMap to get the .WAD even if you're using WinBSPC. Once you've obtained the .WAD, you need to convert it to Source materials. This is done with a program called xwad. Xwad is a command line application, you can get it by downloading the Source SDK on steam, (NOT 2007 or 2013) and navigating the bin folder where the developer tools are stored. You'll need to create a .bat and add the parameters -BaseDir (location of the momentum folder) -WadFile (location of the .wad) and -vtex (automatically calls vtex to convert the created TGA files into VTF files). The .bat would look a little something like this:
+Assuming you are using HUSK MD, after decompiling a .wad file with all the relevant textures will be generated. This .wad file will need to be converted into .vtf and .vmt files so that the textures are usable in source. The easiest way to convert these textures is with VTFEdit.  
+
+Download VTFEdit here: https://github.com/NeilJed/VTFLib  
+  
+With VTFEdit installed, navigate to Tools -> Convert WAD File. Select a .wad to be converted and a relevant folder for the output. Make sure "Create VMT Files" is checked as creating them manually can get quite tedious. The .vmt files generated will not preserve any of the properties of the original textures in the WAD. In other words a glass texture will not be transparent without designating it as such manually in the .vmt file.  
+
+![alt text](https://i.imgur.com/MzowYcc.png)
```

Comment:
Let's host the images directly on the site. This just requires putting this image inside of the `assets/images/` folder somewhere and referencing it by file path (e.g. `![VTF Edit settings](/assets/images/<file path>)`)

Reply:
All of the images are now links to the assets folder. There are the two mentions of "spaghetti" which I do not embed the image for as its lack of importance does not warrant it. As shown below, I link to the asset directly in the repository which opens in a new page. I wanted to specifically point this out to make sure pointing to an image in this manner is alright.

`[spaghetti](/assets/images/goldsrc_to_source_guide/hammer_spaghetti.png)`

Also addressed the following comment from the real hexhexhexhexhexhexhexhexaflexagon.

hexaflexagon — 01/12/2023 8:16 PM
39 to 43 could be markdown links to each project but besides that seems good enough

Reply:
Perfect, thanks!